tools/src/launcher/Main.hs

```diff
@@ -515,11 +519,47 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProc (Process.UseHandle logHandle) path args
```

Comment:
```
            createDirectoryIfMissing True (directory lp)
            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
            let cr = createProc (Process.UseHandle logHandle) path args
```

These four lines are duplicated between `runWallet` and `spawnNode`.
Can you refactor please?